File: php/features.php
<?php
declare(strict_types=1);

// FreeDictionary SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreeDictionaryFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreeDictionaryBaseFeature();
            case "test":
                return new FreeDictionaryTestFeature();
            default:
                return new FreeDictionaryBaseFeature();
        }
    }
}

File: rb/features.rb
# FreeDictionary SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FreeDictionaryFeatures
  def self.make_feature(name)
    case name
    when "base"
      FreeDictionaryBaseFeature.new
    when "ratelimit"
      FreeDictionaryRatelimitFeature.new
    when "retry"
      FreeDictionaryRetryFeature.new
    when "test"
      FreeDictionaryTestFeature.new
    when "timeout"
      FreeDictionaryTimeoutFeature.new
    else
      FreeDictionaryBaseFeature.new
    end
  end
end
